docs/3-cloud-computing/3.2.5-lambda.md

```diff
@@ -0,0 +1,125 @@
+# AWS Lambdas
+
+AWS Lambda is a serverless compute service provided by Amazon Web Services (AWS). It allows developers to run their code without provisioning or managing servers, and only pay for the compute time consumed. Lambdas are event-driven, meaning they are triggered by specific events, such as a user uploading a file to S3 or a change in a DynamoDB table. They can also be used to build and run microservices, perform real-time data processing and analytics, and handle backend tasks such as authentication and authorization. Additionally, Lambdas can be integrated with other AWS services, such as API Gateway, S3, and Kinesis, making it a versatile and powerful tool for building scalable, fault-tolerant applications.
+
+## Exercise - Creating a Lambda function with a trigger
+
+### Overview
+- In this exercise we will be using s3 buckets to upload a csv file to a lambda function, and update a DynamoDB table.
+
+### Steps
+
+1. Set up an s3 bucket.
+
+2. Set up a DynamoDb table
+
+3. Write a python program to grab items from a csv and push to a DynamoDB table
+    - Make sure to have the right permissions
+    - Add a trigger of type s3 bucket
+        - Specify an event type
+            - Hint: Think of what kind of files you will be uploading to the bucket
+
+    - Make a lambda function that grabs every item from a csv file
+
+    Here is some helpful documentation for [boto3](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/dynamodb.html)
+
+``` python
```

Comment:
With both of these code blocks, it might be worth explaining what's being provided, i.e.
> Here's the the scaffolding of lambda code for what we're trying to do. Fill in x,y, and z